edx_rest_framework_extensions/__init__.py

```diff
@@ -1,3 +1,3 @@
 """ edx Django REST Framework extensions. """
 
-__version__ = '7.0.1'  # pragma: no cover
+__version__ = '7.1.0'  # pragma: no cover
```

Comment:
1. I've been wondering whether or not this needs to be considered backward-incompatible. We'll see.
2. On a related compatibility note, someone should understand https://github.com/encode/django-rest-framework/issues/6500 and whether we need any tests around this, or whether it will cause any compatibility changes, etc.

Reply:
@xitij2000: Can you take a quick look at the issue I linked to and see if you think we need any tests, or comments, or code changes, or nothing. While your head is in this work. Thank you!

Reply:
Note: I get that this may be independent, so if you want to punt, that is ok.

Reply:
I did look into the issue, but forgot to update here. 

First, from what I understand, this very same issue exists for "rest_condition" as well. 

The issue in brief is that if you're using object-level permissions, but don't have object-level permissions defined, they default to true from the base class. 

Here is a case where that can be unexpected. 
1. We have a permission `ReadOnly`  that only allows read operation. Here, we've implemented `has_permission`  to be True if the `request.method` is safe method, and didn't implement `has_object_permission` so it's always true (by default).
2. We have a permission like `IsUserCourseAdmin`. Here, `has_permission` can probably return True if the user is a global admin and `has_object_permission` can be true if the user has the right privileges. 

If the above two are compose as `ReadOnly | IsUserCourseAdmin` you'd expect that the read only requests would be allowed for all users, and write requests would be allowed for course admins. 

However, the above permission would **always** resolve to true for all courses. The reason being that our `ReadOnly` permission class **didn't** override the default behaviour of always returning True for `has_object_permission`. 

I think this just means we should be careful when composing permission that apply to objects and make sure that the classing we're composing all specify the behaviour correctly. In the above example the issue could be avoided if you also implemented `has_object_permission` for read only with the same behaviour as `has_permission`.

Reply:
Thanks for this. It is very helpful. 
1. Since we only have a small handful of composed permissions that you are touching across the two PRs (this and edx-platform), do you know if they are all safe to be composed?
2. For `ReadOnly`, although it would be a backward incompatible change, does it make sense to override with comment like was done in https://github.com/encode/django-rest-framework/issues/6500#issuecomment-630175431? Note: I'm just asking the question. I know this change is outside of the scope of this PR.

Reply:
I don't think we have any 'ReadOnly' permission class. Generally very few places use object-level permissions. 

I think I had checked all the permission we've composed in https://github.com/edx/edx-platform/pull/28663 and none of them use object-level permissions. 

I've checked again they don't.

Reply:
Thanks for checking.